src/simulation/ApplyLoad.cpp

```diff
@@ -825,4 +931,181 @@ ApplyLoad::getWriteEntryUtilization()
     return mWriteEntryUtilization;
 }
 
+void
+ApplyLoad::warmAccountCache()
+{
+    auto const& accounts = mTxGenerator.getAccounts();
+    CLOG_INFO(Perf, "Warming account cache with {} accounts.", accounts.size());
+
+    LedgerTxn ltx(mApp.getLedgerTxnRoot());
+    for (auto const& [_, account] : accounts)
+    {
+        auto acc = stellar::loadAccount(ltx, account->getPublicKey());
+        releaseAssert(acc);
+    }
+}
+
+void
+ApplyLoad::findMaxSacTps()
+{
+    releaseAssertOrThrow(mMode == ApplyLoadMode::MAX_SAC_TPS);
+
+    uint32_t minTps = mApp.getConfig().APPLY_LOAD_MAX_SAC_TPS_MIN_TPS;
+    uint32_t maxTps = mApp.getConfig().APPLY_LOAD_MAX_SAC_TPS_MAX_TPS;
+    uint32_t bestTps = 0;
+    double targetCloseTime =
+        mApp.getConfig().APPLY_LOAD_MAX_SAC_TPS_TARGET_CLOSE_TIME_MS;
+
+    CLOG_WARNING(Perf,
+                 "Starting MAX_SAC_TPS binary search between {} and {} TPS",
+                 minTps, maxTps);
+    CLOG_WARNING(Perf, "Target close time: {}ms", targetCloseTime);
+    CLOG_WARNING(Perf, "Num parallel clusters: {}",
+                 mApp.getConfig().APPLY_LOAD_LEDGER_MAX_DEPENDENT_TX_CLUSTERS);
+
+    while (minTps <= maxTps)
+    {
+        uint32_t testTps = (minTps + maxTps) / 2;
```

Comment:
I don't think we need to be so precise here with binary searching the exact TPS, given that there will likely be some noise that will make the function of time to tx count non-monotonic. I think it's at least worth to binary search txs per cluster; I don't think it makes sense to have uneven clusters for the purpose of this test.